src/bionics_ui.cpp

```diff
@@ -724,6 +824,12 @@ void player::power_bionics()
                     popup( _( "You can't toggle auto start mode on a non-fueled CBM." ) );
                 }
             }
+        } else if( action == "SORT" ) {
+            uistate.bionic_sort_mode = pick_sort_mode();
+            // FIXME: is there a better way to resort?
+            active = filtered_bionics( *my_bionics, TAB_ACTIVE );
+            passive = filtered_bionics( *my_bionics, TAB_PASSIVE );
+            g->invalidate_main_ui_adaptor();
```

Comment:
Does this action change information displayed in the main UI (the place where terrain and player etc are displayed)? If not, the call to `invalidate_main_ui_adaptor` is unecessary, as its function is to mark the main UI for redraw.

Reply:
Thanks! I was following the model from the other actions, but it seems to work without that call. I'll commit this with the other suggestions